src/NATS.Client.JetStream/NatsJSContext.cs

```diff
@@ -172,6 +173,26 @@ public NatsJSContext(NatsConnection connection, NatsJSOpts opts)
         throw new NatsJSPublishNoResponseException();
     }
 
+    internal static void ThrowIfInvalidStreamName([NotNull] string? name, [CallerArgumentExpression("name")] string? paramName = null)
+    {
+#if NET8_0_OR_GREATER
+        ArgumentNullException.ThrowIfNullOrEmpty(name);
```

Comment:
looks like net6 and net8 are going to return different messages when the empty.

Reply:
if the name is empty wouldn't they throw totally different exception messages? net8 isn't going to have the custom message set in net6, no?

Reply:
> The message used for .net 6 Is copied from .net 8 framework messages, [this is what .net 8 does](https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/ArgumentException.cs,126) But I should probably change it to use ArrgumentException.ThrowIfNullOrEmpty for .net 8, it's the same static though 😄
> 

ah nice! didn't realise that

> As a side note, I just tried using characters like `\x7 \x8 \x9 \r\n` in different combinations when creating streams, and it gives all kinds of different behaviour , some ends with no response from server, others with the server creating the stream but silently removing the character, and in one case replacing it with an 'x' 😄 so we should probably consider making all the characters below '\1F' "illegal" as well.

What about 'space'? Is the server replacing that with 'x'? Maybe we should stick to minimal checks as someone might be relying on the current behaviour somehow 😅

Reply:
btw, as mentioned in the issue, docs says:
> Names cannot contain whitespace, ., *, >, path separators (forward or backwards slash), and non-printable characters.

https://docs.nats.io/nats-concepts/jetstream/streams#configuration

Reply:
I'm leaning towards less validation (i.e. leave as it is) even though server not responding in some cases is not dev friendly, they should be very rare. cc @Jarema